File: ios/RCTTesseractModule.m
//
//  RCTTesseractModule.m
//  TesseractModule
//
//  Created by Jan Maru De Guzman on 12/22/20.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <TesseractOCR/TesseractOCR.h>

@interface RCT_EXTERN_MODULE(RCTTesseractModule, NSObject)
RCT_EXTERN_METHOD(recognize:(NSString *)uri
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
@end

File: ios/RCTTesseractModule.swift
//
//  RCTTesseractModule.swift
//  TesseractModule
//
//  Created by Jan Maru De Guzman on 12/22/20.
//

import Foundation

@objc(RCTTesseractModule)

class RCTTesseractModule: NSObject, RCTBridgeModule, G8TesseractDelegate {
  static func moduleName() -> String! {
    return "TesseractModule";
  }
  
  static func requiresMainQueueSetup() -> Bool {
    return true;
  }
  
  @objc
  func recognize(_ uri: NSString,
                 resolve: RCTPromiseResolveBlock,
                 reject: RCTPromiseRejectBlock) -> Void {
    let tesseract: G8Tesseract? = G8Tesseract(language: "eng")
    tesseract?.delegate = self
    tesseract?.image = UIImage(named: uri as String)!
    tesseract?.recognize()
    resolve(tesseract?.recognizedText)
  }
  
  func shouldCancelImageRecognitionForTesseract(tesseract: G8Tesseract!) -> Bool {
      return false // return true if you need to interrupt tesseract before it finishes
  }
}
